server/render.js

```diff
@@ -0,0 +1,61 @@
+import * as url from "url";
+import React from "react";
+import ReactDOM from "react-dom/server";
+import fetch from "node-fetch";
+import { getServerStore } from "../src/components/app-state";
+import App from "../src/components/app";
+import { rewind } from "../src/components/use-fetch";
+import { rewind as rewindTitle } from "../src/components/use-title";
+import { rewind as rewindMatchResult } from "../src/lib/router";
+import { getAssetName } from "./serve";
+
+function serializeState(state) {
+  return JSON.stringify(state).replace(/</g, "\\u003c");
+}
+
+export default next => async (req, res) => {
+  const { pathname, search, hash } = url.parse(req.url);
+  const location = { pathname, search, hash };
+  const { cookie } = req.headers;
+  const store = getServerStore();
+
+  const fetchWithCookie = url => fetch(url, { headers: { cookie } });
+
+  ReactDOM.renderToStaticMarkup(
+    <App initialState={store.state} serverLocation={location} />
+  );
```

Comment:
did you forget to send the "React" markup to the client?

Reply:
sorry, just saw your words on this in https://github.com/magjckang/test-fullstack#a-fast-react-server-side-rendering-solution

Reply:
I have a simple explanation in "A Fast React Server Side Rendering Solution" section of `Readme.md`.

Short version is, this site do not need SEO but do need to be fast (it is deployed at Atlanta...).

`ReactDOMServer.renderToStaticMarkup` is even faster than `ReactDOMServer.renderToString` if the client doesn't perform `ReactDOM.hydrate`.

Reply:
Ops... It seems Github has some UX issues.